nomad/drainer/draining_node.go

```diff
@@ -125,6 +125,11 @@ func (n *drainingNode) DrainingJobs() ([]structs.NamespacedID, error) {
 	n.l.RLock()
 	defer n.l.RUnlock()
 
+	// Should never happen
+	if n.node == nil || n.node.DrainStrategy == nil {
+		return nil, fmt.Errorf("node doesn't have a drain strategy set")
```

Comment:
Is returning an error much better than panicking because won't it cause leader flapping?

Reply:
It should be safe how we use it: https://github.com/hashicorp/nomad/blob/master/nomad/drainer/watch_nodes.go#L51-L71

Update will only be called if the node is draining. So it is more going to help future uses not introduce bad logic.